pkg/config/authorization/opa.go

```diff
@@ -19,14 +21,24 @@ default allow = false
 	policyUIDHashSeparator = "|"
 
 	opaPolicyPrecompileErrorMsg = "Failed to precompile OPA policy"
+	regoDownloadErrorMsg        = "Failed to download Rego data"
 	invalidOPAResponseErrorMsg  = "Invalid response from OPA policy evaluation"
 )
 
 var (
-	opaLog = ctrl.Log.WithName("Authorino").WithName("ApiKey")
+	opaLog = ctrl.Log.WithName("Authorino").WithName("OPA")
 )
 
-func NewOPAAuthorization(policyName string, rego string, nonce int) *OPA {
+func NewOPAAuthorization(policyName string, rego string, url string, nonce int) *OPA {
+	if rego == "" && url != "" {
+		downloadedRego, err := downloadRegoDataFromUrl(url)
+		if err != nil {
+			opaLog.Error(err, regoDownloadErrorMsg, "secret", policyName)
+			return nil
```

Comment:
I think we should return the error in this case, as fetching the policy is essential.

Reply:
What we will do with the error in the service controller? Fail the reconciliation? 
https://github.com/Kuadrant/authorino/blob/9b9b6d7fe742ba90a2af54fbdb2bfffc81d6141f/controllers/service_controller.go#L232

Reply:
It's either that or trying again before processing the first request.

For OIDC identity, for example, if the OpenID Connect config cannot be fetched in reconciliation time, we try again in request-time. We could do the same here.

...Or as I suggested before. We modify `NewOPAAuthorization` to return `(*OPA, error)` and let the controller to treat it when there's an error (e.g. retry reconciling the resource).

Reply:
> For OIDC identity, for example, if the OpenID Connect config cannot be fetched in reconciliation time, we try again in request-time. We could do the same here. 

Do you mean inside the `call` function?  


I would go for both options because I was missing the error when my `InlineRego` was faulty.

Reply:
> Do you mean inside the `call` function?

Yes.

Reply:
Returning the error is acceptable, but it's either one thing or the other.

If we reject creating the config (because we cannot fetch the policy, or because compiling the policy failed), then so the reconciler should fail and fall back to its default, which is retrying to reconcile the resource later.

On the other hand, if we chose for accepting the config silently, the user will only realize something is wrong when a first request is sent. In the case of OIDC identity, this was my only option because I wanted to support it in combination with wristband configs in the same custom resource.

Reply:
Then I would go for failing the reconciliation. It will useful for both options.